sql/src/test/java/org/apache/druid/sql/calcite/CalciteNestedDataQueryTest.java

```diff
@@ -2692,9 +2692,12 @@ public void testGroupByInvalidPath()
         + "SUM(cnt) "
         + "FROM druid.nested GROUP BY 1",
         (expected) -> {
-          expected.expect(UnsupportedSQLQueryException.class);
+          expected.expect(DruidException.class);
           expected.expectMessage(
-              "Cannot use [JSON_VALUE_VARCHAR]: [Bad format, '.array.[1]' is not a valid JSONPath path: must start with '$']");
+              "Cannot use JSON_VALUE_VARCHAR\n" +
+              "Error Code: Unsupported operation\n" +
+              "Cause: Bad format, '.array.[1]' is not a valid JSONPath path: must start with '$'"
```

Comment:
This change in test highlights something taht we definitely don't want to happen: the message of the exception is losing valuable information about what actual bad thing happened.  On top of that, if we happen to be serializing it out with all of the context, then we get the information back, but now it's new-line delimited and new-line delimited exception messages are very annoying to deal with when trying to operate a system.

Instead of generating a new-line delimited error message, when we are actively trying to log the full context, it should all be on the same line as the original error message.  And, the error message itself should be considered a self-contained, meaningful thing that can be used by its intended audience without anything from the context needing to also exist.

Reply:
Reverted the change. We can focus our efforts where we have poor errors. This one seems fine as is.